lib/symboldatabase.cpp

```diff
@@ -6018,7 +6018,10 @@ const Function* Scope::findFunction(const Token *tok, bool requireConst, Referen
         matches.erase(itPure);
 
     // Only one candidate left
-    if (matches.size() == 1)
+    if (matches.size() == 1 && std::none_of(functionList.begin(), functionList.end(), [tok](const Function& f) {
+        const auto nTok = tok->str().size();
+        return startsWith(f.name(), tok->str()) && f.name().size() > nTok + 2 && f.name()[nTok + 1] == '<';
```

Comment:
```suggestion
        return startsWith(f.name(), tok->str() + "<") && f.name().size() > nTok + 2;
```
I still feel it's hacky but this seems better.

Reply:
The function name seems to be `func < arg >` instead of `func<arg>` for some reason.

Reply:
yeah there is (sometimes?) space before the "<" . we did not used to have it. maybe the space is removed in the future again who knows.

Reply:
If you have examples for either case I will see if I can bisect this.

Reply:
> If you have examples for either case I will see if I can bisect this.

The spaces appear in the output of `--debug -v` (e.g. `name:`), but I don't know what the significance is.

Reply:
I assume you mean this:
```
Scope: 00000237081B6C40 Function
    className: fun < std :: vector < std :: string > >
```

There were always whitespaces in that since that function appeared in version 1.84. I guess before that there were no template instantiations.

Reply:
```cpp
template <class T> void fun()
{
}

int main()
{
    fun<int>();
}
```

1.77
```
##file S:/___temp/input.cpp

1:
|
4:
5: int main ( )
6: {
7: fun<int> ( ) ;
8: } void fun<int> ( )
2: {
3: }
[...]
Scope: 000001F7E6853E30 Function
    className: fun<int>
[...]
##AST
( 'signed int'
`-main

( 'void'
`-fun<int>

( 'void'
`-fun<int>
```

1.78
```
##file S:/___temp/input.cpp

1:
|
4:
5: int main ( )
6: {
7: fun < int > ( ) ;
8: } void fun < int > ( )
2: {
3: }
[...]
Scope: 000001CBB41650D0 Function
    className: fun < int >
[...]
##AST
( 'signed int'
`-main

( 'void'
`-fun < int >

( 'void'
`-fun < int >
```

1.82
```
1:
|
4:
5: int main ( )
6: {
7: fun<int> ( ) ;
8: } void fun<int> ( )
2: {
3: }
[...]
Scope: 000001E34CC268D0 Function
    className: fun < int >
[...]
##AST
( 'signed int'
`-main

( 'void'
`-fun < int >

( 'void'
`-fun < int >
```

Reply:
Something in here we need to file a ticket about?

Reply:
The inconsistent use of spaces in template names, perhaps.

Reply:
https://trac.cppcheck.net/ticket/13781